packages/manager/modules/exchange/src/account/alias/add/account-alias-add.controller.js

```diff
@@ -108,17 +122,25 @@ export default class ExchangeAddAccountAliasCtrl {
       });
   }
 
-  aliasIsValid() {
-    if (isEmpty(this.model)) {
-      return false;
+  checkAliasValidity() {
+    let isAliasValid;
+    if (this.model?.radio === this.ALIAS_TYPE.smtp) {
+      const aliasIsValid =
+        !isEmpty(this.model.alias) &&
+        isString(this.model.alias) &&
+        this.model.alias.length <= 64;
+      const hasDomain = !isEmpty(this.model.domain);
+      isAliasValid = aliasIsValid && hasDomain && !this.takenEmailError;
+    } else {
+      const match = this.model.x500.match(X500_REGEX) || [];
+      isAliasValid = match.length === 1 && match[0] === this.model.x500;
     }
+    return isAliasValid;
```

Comment:
I think we could do that:

```suggestion
    
    if (this.model?.radio === this.ALIAS_TYPE.smtp) {
      const aliasIsValid =
        !isEmpty(this.model.alias) &&
        isString(this.model.alias) &&
        this.model.alias.length <= 64;
      const hasDomain = !isEmpty(this.model.domain);
      return= aliasIsValid && hasDomain && !this.takenEmailError;
    } 
     
     const match = this.model.x500.match(X500_REGEX) || [];
     return match.length === 1 && match[0] === this.model.x500;
   
```

Reply:
I totally agree with this suggestion. But I was getting a lint error for putting statements after 1st `return`